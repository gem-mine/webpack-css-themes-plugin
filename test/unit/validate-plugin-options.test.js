const MiniCssExtractPlugin = require('../../src')

describe('validate options', () => {
  const tests = [{
    key: 'full options',
    type: 'success',
    value: {
      themes: [{
        name: 'xxx',
        filePath: 'testPath'
      }],
      minify: false,
      isCSSModules: true
    },
  }, {
    key: 'no boolean',
    type: 'success',
    value: {
      themes: [{
        name: 'xxx',
        filePath: 'testPath'
      }],
    }
  }, {
    key: 'no theme',
    type: 'failure',
    value: {}
  }, {
    key: 'empty theme',
    type: 'failure',
    value: {
      themes: []
    }
  }, {
    key: 'error minify type',
    type: 'failure',
    value: {
      themes: [{
        name: 'xxx',
        filePath: 'testPath'
      }],
      minify: 'xx',
    }
  }]

  function createTestCase(key, value, type) {
    it(`should ${
      type === 'success' ? 'successfully validate' : 'throw an error on'
    } the "${key}" option with "${JSON.stringify(value)}" value`, () => {
      let error

      try {
        // eslint-disable-next-line no-new
        new MiniCssExtractPlugin(value)
      } catch (errorFromPlugin) {
        if (errorFromPlugin.name !== 'ValidationError') {
          throw errorFromPlugin
        }

        error = errorFromPlugin
      } finally {
        if (type === 'success') {
          expect(error).toBeUndefined()
        } else if (type === 'failure') {
          expect(() => {
            throw error
          }).toThrowErrorMatchingSnapshot()
        }
      }
    })
  }

  tests.forEach(({
    key, value, type
  }) => {
    createTestCase(key, value, type)
  })
})
