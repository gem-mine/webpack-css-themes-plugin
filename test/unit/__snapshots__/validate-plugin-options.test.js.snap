// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`validate options should throw an error on the "empty theme" option with "{"themes":[]}" value 1`] = `
"Invalid configuration object. Object has been initialized using a configuration object that does not match the API schema.
 - configuration.themes should be an non-empty array."
`;

exports[`validate options should throw an error on the "error minify type" option with "{"themes":[{"name":"xxx","fileName":"testPath"}],"minify":"xx"}" value 1`] = `
"Invalid configuration object. Object has been initialized using a configuration object that does not match the API schema.
 - configuration.minify should be a boolean."
`;

exports[`validate options should throw an error on the "no theme" option with "{}" value 1`] = `
"Invalid configuration object. Object has been initialized using a configuration object that does not match the API schema.
 - configuration misses the property 'themes'. Should be:
   [object { name?, fileName? }, ...] (should not have fewer than 1 item)"
`;
